examples/ct/structures.go

```diff
@@ -99,15 +112,208 @@ func SignV1SCTForPrecertificate(km crypto.KeyManager, cert *x509.Certificate, t
 	keyHash := sha256.Sum256(cert.RawSubjectPublicKeyInfo)
 	precert := ct.PreCert{IssuerKeyHash: keyHash, TBSCertificate: cert.RawTBSCertificate}
 
-	timestampedEntry := ct.TimestampedEntry{Timestamp:sctInput.Timestamp, EntryType:ct.PrecertLogEntryType, PrecertEntry:precert}
-	leaf := ct.MerkleTreeLeaf{Version: ct.V1, LeafType:ct.TimestampedEntryLeafType, TimestampedEntry:timestampedEntry}
+	timestampedEntry := ct.TimestampedEntry{Timestamp: sctInput.Timestamp, EntryType: ct.PrecertLogEntryType, PrecertEntry: precert}
+	leaf := ct.MerkleTreeLeaf{Version: ct.V1, LeafType: ct.TimestampedEntryLeafType, TimestampedEntry: timestampedEntry}
 
 	return serializeAndSignSCT(km, leaf, sctInput, t)
 }
 
-func getSCTForSignatureInput(t time.Time) (ct.SignedCertificateTimestamp) {
+func getSCTForSignatureInput(t time.Time) ct.SignedCertificateTimestamp {
 	return ct.SignedCertificateTimestamp{
-		SCTVersion:ct.V1,
+		SCTVersion: ct.V1,
 		Timestamp:  uint64(t.UnixNano() / millisPerNano), // spec uses millisecond timestamps
 		Extensions: ct.CTExtensions{}}
-}
\ No newline at end of file
+}
+
+func NewLogEntry(leaf ct.MerkleTreeLeaf, certChain []*x509.Certificate) *CTLogEntry {
+	chain := []ct.ASN1Cert{}
+
+	for _, cert := range certChain {
+		chain = append(chain, cert.Raw)
+	}
+
+	logEntry := new(CTLogEntry)
```

Comment:
You might more concisely write:

return &CTLogEntry{Leaf: leaf, Chain: chain}

replacing these 6 lines.